lib/mpi/messages/find_profile_message_with_address.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require_relative 'find_profile_message_fields'
+require_relative 'find_profile_message_with_address_fields'
+require_relative 'find_profile_message_helpers'
+require 'mpi/constants'
+
+module MPI
+  module Messages
+    class FindProfileMessageWithAddress
+      include FindProfileMessageHelpers
+      attr_reader :address, :given_names, :family_name, :birth_date, :gender, :search_type
+
+      def initialize(profile)
+        required_fields_present?(profile)
+        @given_names = profile[:given_names]
+        @family_name = profile[:last_name]
+        @birth_date = profile[:birth_date]
+        # gender is optional and will default to nil if it DNE
+        @gender = profile[:gender]
+
+        @address = profile[:address]
+        @street_address_lines = [address['addressLine1'], address['addressLine2'], address['addressLine3']].compact
```

Comment:
we should be using symbols instead of strings inside the address hash, otherwise the profile hash gets really confusing (mixes symbols and strings)